micrometer-core/src/main/java/io/micrometer/core/instrument/binder/httpcomponents/PoolingHttpClientConnectionManagerMetricsBinder.java

```diff
@@ -0,0 +1,222 @@
+            .tags(tags).tag(TAG_CONNECTIONS_STATE, "available")
+            .register(registry);
+        poolRouteLeasedGauge = MultiGauge.builder(METER_ROUTE_CONNECTIONS)
+            .description(METER_ROUTE_CONNECTIONS_DESC)
+            .tags(tags).tag(TAG_CONNECTIONS_STATE, "leased")
+            .register(registry);
+        poolRoutePendingGauge = MultiGauge.builder(METER_ROUTE_PENDING)
+            .description(METER_ROUTE_PENDING_DESC)
+            .tags(tags)
+            .register(registry);
+    }
+
+    /**
+     * Updates the list of routes accessed by the monitored http client.
+     *
+     * Call this method periodically if you want to monitor thread pools for
+     * every accessed route (target scheme, host and port). For example:
+     *
+     * <pre>
+     *     &#x40;Bean
+     *     public RouteUpdater routeUpdater(List&#60;PoolingHttpClientConnectionManagerMetricsBinder&#62; metricsBinders) {
+     *         return new RouteUpdater(metricsBinders);
+     *     }
+     *
+     *     private static class RouteUpdater {
+     *         private final HashSet&#60;PoolingHttpClientConnectionManagerMetricsBinder&#62; metricsBinders = new HashSet&#60;&#62;();
+     *
+     *         private RouteUpdater(Collection&#60;PoolingHttpClientConnectionManagerMetricsBinder&#62; metricsBinders) {
+     *             this.metricsBinders.addAll(metricsBinders);
+     *         }
+     *
+     *         &#x40;Scheduled(fixedRate = 10000)
+     *         public void updateRoutes() {
+     *             metricsBinders.forEach(PoolingHttpClientConnectionManagerMetricsBinder::updateRoutes);
+     *         }
+     *     }
+     * </pre>
+     *
+     * BE CAREFUL! Call this if and only if you are sure that your http client
+     * accesses only a limited number of remote schemes, hosts and ports. DO NOT
+     * USE THIS METHOD if your HttpClient accesses a big or unlimited number of
+     * routes (i.e. if the target host depends on the user input).
+     */
+    @SuppressWarnings("WeakerAccess")
+    public void updateRoutes() {
+        Set<HttpRoute> routes = connectionManager.getRoutes();
+        poolRouteMaxGauge.register(routesToRows(routes, PoolStats::getMax));
+        poolRouteAvailableGauge.register(routesToRows(routes, PoolStats::getAvailable));
+        poolRouteLeasedGauge.register(routesToRows(routes, PoolStats::getLeased));
+        poolRoutePendingGauge.register(routesToRows(routes, PoolStats::getPending));
+    }
+
+    private Iterable<MultiGauge.Row<?>> routesToRows(Set<HttpRoute> routes, Function<PoolStats, Integer> valueFunction) {
+        return routes.stream()
+            .map((route) -> routeToRow(route, () -> valueFunction.apply(connectionManager.getStats(route))))
+            .collect(Collectors.toSet());
+    }
+
+    private MultiGauge.Row<Supplier<Number>> routeToRow(HttpRoute route, Supplier<Number> valueFunction) {
+        Tags tags = Tags.of(
+            "target.host", route.getTargetHost().getHostName(),
```

Comment:
I don't feel strongly about it, but `target` feels a little redundant since this is a tag on metrics for an HTTP client. That is, without `target` it might be clear enough what is meant, given the context. Anyone else have an opinion on it?

Reply:
With a simple `host` tag it might get confusing if you tag all (non-http-client-related) metrics with the hostname of your own service ?  For example, I have multiple instances of my service and I tag all the metrics with `host` = hostname to differentiate between my instances.

Then again, OkHttpClient instrumentation seems to be using `target` too.

Reply:
Good point. Let's keep as is for now, then.